validation.py

```diff
@@ -0,0 +1,492 @@
+                max_ = max(np.nanmax(x), np.nanmax(y))
+                ax.plot([min_, max_], [min_, max_],
+                        c="lightgray", ls="--", lw=1, zorder=-10)
+
+            # use a full-figure subplot for labels that span the other subplots
+            ax = fig.add_subplot(111, frameon=False)
+            ax.set_xticks([])
+            ax.set_yticks([])
+            ax.set_title(f"{sim_key1} vs. {sim_key2}: {stat_key}", pad=30)
+            ax.set_xlabel(sim_key1, labelpad=30)
+            ax.set_ylabel(sim_key2, labelpad=50)
+            fig.tight_layout()
+            pdf.savefig(figure=fig, bbox_inches='tight')
+            plt.close(fig)
+
+        pdf.close()
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+            description="Do validation simulations and make QQ plots.")
+    parser.add_argument(
+            "-o", "--output-folder", metavar="DIR", default="validation",
+            help="Folder to store validation plots and tree sequences "
+                 "[%(default)s].")
+
+    mutex_group = parser.add_mutually_exclusive_group()
+    mutex_group.add_argument(
+            "-n", "--no-plots", action="store_true", default=False,
+            help="Don't make plots, just do the simulations [%(default)s].")
+    mutex_group.add_argument(
+            "-p", "--plot-only", action="store_true", default=False,
+            help="Don't simulate, just make QQ plots from preexisting files "
+                 "[%(default)s].")
+
+    parser.add_argument(
+            "-j", "--num-procs", metavar="NPROCS", type=int, default=1,
+            help="Number of simulations to run simultaneously [%(default)s].")
+    parser.add_argument(
+            "-r", "--num-replicates", metavar="NREPS", type=int, default=100,
+            help="Number of replicates for each simulation key [%(default)s].")
+    parser.add_argument(
+            "-s", "--seed", metavar="SEED", type=int, default=1234,
+            help="Seed for the random number generator [%(default)s].")
+    parser.add_argument(
+            "keys", nargs="*",
+            help="One or more scenarios to simulate and/or compare.")
+
+    args = parser.parse_args()
+
+    if len(args.keys) == 0:
+        args.comparisons = default_comparisons
+        args.keys = list(set(itertools.chain(*args.comparisons)))
+    else:
+        args.comparisons = itertools.combinations(args.keys, 2)
+
+    # sort keys to get deterministic ordering from random number generator
+    args.keys.sort()
+
+    for key in args.keys:
+        if key not in simulation_functions:
+            if args.plot_only:
+                # Might be a mistake, but continue anyway to allow validation
+                # using arbitrary folders that are in the right place.
+                warning(f"unknown scenario key ``{key}''")
+            else:
+                parser.error(f"unknown scenario key ``{key}''")
+
+    return args
+
+
+if __name__ == "__main__":
+    args = parse_args()
```

Comment:
I think it's better to split the job up explicitly into "run" and "plot" (or some other better names). The running bit will end up taking a lot of time, and you want to be able to do the two things independently. It's especially useful to be able to run plots for simulations that are ongoing (so it's good to always flush out any results that have been calculated when you can).